ui/src/shared/utils/fromFlux.ts

```diff
@@ -0,0 +1,345 @@
+                 |                   |          |               1.0 |     true
+                 |                   |          |               2.0 |     true
+                 |                   |          |               3.0 |     true
+
+    The `#group`, `#datatype`, and `#default` [annotations][1] are expected to be
+    in the input Flux CSV.
+    Values are coerced into appropriate JavaScript types based on the Flux
+    `#datatype` annotation for the table
+    The `Table` stores a `key` for each column which is separate from the column
+    `name`. If multiple Flux tables have the same column but with different
+    types, they will be distinguished by different keys in the resulting `Table`;
+    otherwise the `key` and `name` for each column in the result table will be
+    identical.
+
+    [0]: https://github.com/influxdata/flux/blob/master/docs/SPEC.md#csv
+    [1]: https://github.com/influxdata/flux/blob/master/docs/SPEC.md#annotations
+    */
+
+interface ChunkBoundary {
+  start: number
+  stop: number
+}
+
+type ParsedFluxColumnLookup = {[key: string]: ParsedFluxColumn}
+
+export interface ParsedFluxColumn {
+  name: string
+  group: string
+  type: InternalTypes
+  default: string
+  data: ParsedReturnTypes[]
+}
+
+export interface ParsedFlux {
+  table: {
+    columnKeys: string[]
+    columns: ParsedFluxColumnLookup
+    length: number
+  }
+  fluxGroupKeyUnion: string[]
+}
+
+export default function fromFlux(csv: string): ParsedFlux {
+  /*
+         A Flux CSV response can contain multiple CSV files each joined by a newline.
+         See https://github.com/influxdata/flux/blob/master/docs/SPEC.md#multiple-tables.
+
+         Split the response into separate chunks whenever we encounter:
+             1. A newline
+             2. Followed by any amount of whitespace
+             3. Followed by a newline
+             4. Followed by a `#` character
+         The last condition is [necessary][0] for handling CSV responses with
+         values containing newlines.
+
+         [0]: https://github.com/influxdata/influxdb/issues/15017
+     */
+  const output: ParsedFluxColumnLookup = {},
+    groupKey: {[key: string]: boolean} = {},
+    names: {[key: string]: {[key: string]: boolean}} = {}
+
+  const chunks: ChunkBoundary[] = [],
+    regerz = /\n\s*\n#/g
+  let match,
+    lastRange = 0
+
+  while ((match = regerz.exec(csv)) !== null) {
+    chunks.push({
+      start: lastRange,
+      stop: match.index,
+    })
+    lastRange = match.index + match[0].length - 1
+  }
+  chunks.push({
+    start: lastRange,
+    stop: csv.length,
+  })
+
+  let runningTotal = 0,
+    currentChunkIndex,
+    headerLocation,
+    currentLineInHeader,
+    currentColumnIndex,
+    currentLineIndex,
+    columnName,
+    columnType,
+    columnKey,
+    annotations,
+    parsed
+
+  for (
+    currentChunkIndex = 0;
+    currentChunkIndex < chunks.length;
+    currentChunkIndex++
+  ) {
+    annotations = {}
+    parsed = Papa.parse(
```

Comment:
i hear library developers love it when you don't rename when declaring a library's default export definition as big papa.
:rip biggie: